middle_end/flambda2/simplify/env/downwards_env.ml

```diff
@@ -47,14 +47,15 @@ type t =
     all_code : Code.t Code_id.Map.t;
     inlining_history_tracker : Inlining_history.Tracker.t;
     loopify_state : Loopify_state.t;
-    continuation_stack : Continuation.t list;
-    variables_defined_in_current_continuation : Lifted_cont_params.t;
-    variables_defined_in_current_continuation' : Lifted_cont_params.t;
     lifted : Variable.Set.t;
-    number_of_continuations_defined_in_current_continuation : int
-        (* this last field makes it possible to count the number of parameters
-           added by the lifting process, and therefore estimate the cost of
-           lifting continuations. *)
+    defined_variables : Lifted_cont_params.t list;
```

Comment:
Maybe `defined_variables_by_scope`, as that reflects a bit better why we're organising them into a list. Also, the comment should explain why we're storing `Lifted_cont_params.t` (basically because it's only used for lifting so that's the more convenient type).